### DISCUSSION TITLE
What do "processing strength" and "fix frame strength" do?

### DISCUSSION THREAD

**Discussion by afgonczol** [volotat/SD-CN-Animation#43] [Q&A] [answered]
What do "processing strength" and "fix frame strength" do?

**Comment by Squeezitgirdle** [comment]
Far as I can tell, a low "fix frame strength" is like setting denoise strength in img2img to 0.
Setting it higher makes larger changes to the image.

Processing strength somewhat seems to do that too, but not quite? I haven't quite figured it out yet, though so hoping for a better answer here.

**Comment by volotat** [comment, marked as answer]
With some simplifications, in both vid2vid and txt2vid cases the algorithm works with two img2img passes for each frame. First pass will restore occluded parts of warped previous frame and the second pass will process the whole image to fix any issues that are left after the first step. Technically the second pass is not necessary, but without it frames tend to blur out and have a lot of ghosting artifacts. "processing strength" is a denoising strength for first img2img pass and "fix frame strength" is a denoising strength for second img2img pass. "fix frame strength" should not be too high or too low. I found out that 0.15 works the best for vid2vid mode and 0.35 for txt2vid that's why they are set by default.

**Reply by Squeezitgirdle** [reply]
Thanks for the response, I noticed when I leave it on default settings though. Lets say I use a gif for example such as this one I used for testing purposes since it was so simple:
![62-anime-dance-girl-cat-usagif](https://github.com/volotat/SD-CN-Animation/assets/20117466/02370197-9637-45a6-8d9b-f54b65517f97)

By the time the gif is complete, it usually makes the overlay character kind of transparent until halfway through the gif it's just the original character in the gif. This is much more apparent on longer gifs than this one. 

Additionally I get a lot of random images inserted into the gif that are completely different images than the original gif.
Perhaps its because I'm using anime models?

**Reply by boundlessliving** [reply]
I had the same thing. The Vid2Vid started great with a face replacement, but soon, everything faded back to the original face? Can we keep that from happening like the Jessica Chastain example?

**Reply by volotat** [reply]
I have plans to make some testing examples that will be necessary to see the correctness of the working pipeline and maybe to measure some quality metrics. It most likely will be done after the API and reproducibility will be built. An example with Jessica was produced in v0.4 if I remember correctly, there might be some changes in vid2vid accumulating since then.

**Reply by dianacromatica** [reply]
Is there any way to find the settings on past vid2vid generated? I love th result of one I did but I changed all the parameters, and I don't  remember the settings
